integration_test/srchydration/src_hydration_test.go

```diff
@@ -0,0 +1,354 @@
+package srchydration
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource/transformer"
+
+	"github.com/rudderlabs/rudder-go-kit/jsonparser"
+
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+
+	"github.com/rudderlabs/rudder-server/testhelper/backendconfigtest"
+	webhookutil "github.com/rudderlabs/rudder-server/testhelper/webhook"
+
+	"github.com/samber/lo"
+
+	proctypes "github.com/rudderlabs/rudder-server/processor/types"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	kithttputil "github.com/rudderlabs/rudder-go-kit/httputil"
+	kithelper "github.com/rudderlabs/rudder-go-kit/testhelper"
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource/postgres"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/runner"
+	"github.com/rudderlabs/rudder-server/testhelper/health"
+	"github.com/rudderlabs/rudder-server/testhelper/transformertest"
+)
+
+const (
+	writeKey    = "writekey-1"
+	sourceID    = "source-1"
+	workspaceID = "workspace-1"
+	srcDefName  = "Source-Def-Name-1"
+)
+
+func TestSrcHydration(t *testing.T) {
```

Comment:
Can we run the same scenarios for all isolation modes (none, workspace, source)?
Can we have more than 1 source as well?